src/javascripts/Runners/assetHandlers/scriptFragmentHandler.js

```diff
@@ -67,23 +73,60 @@ ngapp.run(function(mergeAssetService, assetHelpers, pexService, progressLogger,
         script.stringTable[0] = fileName;
         fh.jetpack.dir(fh.getDirectory(newPath));
         pexService.saveScript(script, newPath);
-        xelib.SetValue(a.handle, '', fileName);
+        let oldFileName = xelib.GetValue(a.handle, `Script Fragments\\fileName`);
+        log(`Changing fragment fileName from ${oldFileName} to ${fileName}`);
+        xelib.SetValue(a.handle, 'Script Fragments\\fileName', fileName);
+        let fragmentHandle = xelib.GetElement(a.handle, 'Script Fragments\\Fragments');
```

Comment:
this looks like you're building in handling for when there's an array of fragments vs. a single fragment, but you're calling the `SetValue` even if an array of fragments is present.  the code paths should be explicit and separate so there aren't any "dud" operations which are meaningless.

Reply:
I'm not sure what you mean here?

Line 78 sets the file name, while 79+ sets the script name for each fragment in the array?

Reply:
is that how it works?  there's a `fileName` property at `Script Fragments\\fileName` and within each `Fragment` in the `Script Fragments\\Fragments` array?

Reply:
Yep (although it's scriptName), these are the paths we have to check for and change:
`VMAD\\Script Fragments\\fileName`
`VMAD\\Script Fragments\\Fragments\\scriptName`
`VMAD\\Scripts\\Script\\scriptName`